app/src/main/java/com/swent/assos/ui/screens/manageAsso/CreateEvent.kt

```diff
@@ -1,25 +1,183 @@
 import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateListOf
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.ColorFilter
 import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.semantics.CustomAccessibilityAction
+import androidx.compose.ui.semantics.clearAndSetSemantics
+import androidx.compose.ui.semantics.customActions
+import androidx.compose.ui.semantics.semantics
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import androidx.hilt.navigation.compose.hiltViewModel
+import com.maxkeppeker.sheets.core.models.base.SelectionButton
+import com.maxkeppeker.sheets.core.models.base.rememberUseCaseState
+import com.maxkeppeler.sheets.date_time.DateTimeDialog
+import com.maxkeppeler.sheets.date_time.models.DateTimeSelection
+import com.swent.assos.R
+import com.swent.assos.model.data.Event
+import com.swent.assos.model.data.EventField
+import com.swent.assos.model.data.EventFieldImage
+import com.swent.assos.model.data.EventFieldText
+import com.swent.assos.model.data.EventFieldType
+import com.swent.assos.model.generateUniqueID
 import com.swent.assos.model.navigation.NavigationActions
+import com.swent.assos.model.view.AssoViewModel
+import java.time.LocalDateTime
+import java.time.LocalTime
+import java.time.format.DateTimeFormatter
+import java.time.format.FormatStyle
+import sh.calvin.reorderable.ReorderableItem
+import sh.calvin.reorderable.rememberReorderableLazyColumnState
 
 @OptIn(ExperimentalMaterial3Api::class)
 @Composable
 fun CreateEvent(assoId: String, navigationActions: NavigationActions) {
+
+  val viewModel: AssoViewModel = hiltViewModel()
+
+  val listFieldsReordered = remember { mutableStateListOf<EventField>() }
+  val listFieldsOriginal = remember { mutableStateListOf<EventField>() }
+  var openAlertDialogAddFields by remember { mutableStateOf(false) }
+  var openAlertDialogFields by remember { mutableStateOf(false) }
+  var currentFieldType: EventFieldType by remember { mutableStateOf(EventFieldType.TEXT) }
+
+  var titleEvent by remember { mutableStateOf("") }
+  var description by remember { mutableStateOf("") }
+  var imageURL by remember { mutableStateOf("") }
+  var showTimePickerStart by remember { mutableStateOf(false) }
+  var showTimePickerEnd by remember { mutableStateOf(false) }
+  var startTime by remember { mutableStateOf<LocalDateTime?>(null) }
+  var startHourFormat by remember { mutableStateOf(HourFormat.AM) }
+  var endTime by remember { mutableStateOf<LocalDateTime?>(null) }
+  var endHourFormat by remember { mutableStateOf(HourFormat.AM) }
```

Comment:
Maybe you should move these states into the ViewModel, which would allow for the creation of the event or other non-UI code to be directly handled in the ViewModel. This would make everything clearer and properly separate the UI from state management and backend handling.